src/kvm.c

```diff
@@ -331,11 +331,13 @@ kvm_ioctl_run_attach_auxstr(struct tcb *const tcp,
 
 {
 	static struct kvm_run vcpu_run_struct;
+	kernel_ulong_t mmap_addr;
 
 	if (info->mmap_len < sizeof(vcpu_run_struct))
 		return;
 
-	if (umove(tcp, info->mmap_addr, &vcpu_run_struct) < 0)
+	 mmap_addr = truncate_kulong_to_current_wordsize(info->mmap_addr);
+	if (umove(tcp, >mmap_addr, &vcpu_run_struct) < 0)
```

Comment:
I bet you mean `mmap_addr`, not `>mmap_addr`.

Reply:
Thank you.

Reply:
By the way, `mmap_addr` field should rather have some unsigned type, otherwise there could be a sign extension issue in this umove() invocation.

Reply:
Also, I don't think truncate_kulong_to_current_wordsize is needed here because the argument doesn't have a kernel_ulong_t type.  However, the type of umoven's argument is kernel_ulong_t, so my bet is that the issue will be fixed as soon as the type of mmap_addr field is fixed.

Reply:
I see. I will prepend such a commit to this pull request.
I wonder why I chose long for the purpose...